testing/test_config.py

```diff
@@ -583,3 +583,60 @@ def test_with_specific_inifile(self, tmpdir):
         inifile = tmpdir.ensure("pytest.ini")
         rootdir, inifile, inicfg = determine_setup(inifile, [tmpdir])
         assert rootdir == tmpdir
+
+class TestOverrideIniArgs:
+    """ test --override-ini """
+    @pytest.mark.parametrize("name", "setup.cfg tox.ini pytest.ini".split())
+    def test_override_ini_names(self, testdir, name):
+        testdir.tmpdir.join(name).write(py.std.textwrap.dedent("""
+            [pytest]
+            custom = 1.0
+        """))
+        testdir.makeconftest("""
+            def pytest_addoption(parser):
+                parser.addini("custom", "")
+        """)
+        testdir.makepyfile("""
+            import pytest
+            def test_pass():
+                ini_val = pytest.config.getini("custom")
+                print('\\ncustom_option:%s\\n' % ini_val)
+        """)
+
+        result = testdir.runpytest("--override-ini", "custom=2.0", "-s")
+        assert result.ret == 0
+        result.stdout.fnmatch_lines([
+            "custom_option:2.0"
+        ])
+
+        result = testdir.runpytest("--override-ini", "custom=2.0",
+                                   "--override-ini=custom=3.0", "-s")
+        assert result.ret == 0
+        result.stdout.fnmatch_lines([
+            "custom_option:3.0"
+        ])
+
+
+    def test_override_ini_pathlist(self, testdir):
+        testdir.makeconftest("""
+                def pytest_addoption(parser):
+                    parser.addini("paths", "my new ini value", type="pathlist")
+        """)
+        testdir.makeini("""
+                [pytest]
+                paths=blah.py
+        """)
+        testdir.makepyfile("""
+                import pytest
+                import py.path
+                def test_pathlist():
+                    config_paths = pytest.config.getini("paths")
+                    print(config_paths)
+                    for cpf in config_paths:
+                        print('\\nuser_path:%s' % cpf.basename)
+        """)
+        result = testdir.runpytest("--override-ini", 'paths=foo/bar1.py foo/bar2.py', "-s")
+        result.stdout.fnmatch_lines([
+            "user_path:bar1.py",
+            "user_path:bar2.py"
+        ])
```

Comment:
Could you add a test which sets two different options (say, `custom_option_1` and `custom_option_2` via `--override-ini`?